discovery/reliable_sender.go

```diff
@@ -131,18 +131,23 @@ spawnHandler:
 // spawnPeerMsgHandler spawns a peerHandler for the given peer if there isn't
 // one already active. The boolean returned signals whether there was already
 // one active or not.
-func (s *reliableSender) spawnPeerHandler(peerPubKey [33]byte) (peerManager, bool) {
-	s.activePeersMtx.Lock()
-	defer s.activePeersMtx.Unlock()
+func (s *reliableSender) spawnPeerHandler(
+	peerPubKey [33]byte) (peerManager, bool) {
 
+	s.activePeersMtx.Lock()
 	msgHandler, ok := s.activePeers[peerPubKey]
 	if !ok {
 		msgHandler = peerManager{
 			msgs: make(chan lnwire.Message),
 			done: make(chan struct{}),
 		}
 		s.activePeers[peerPubKey] = msgHandler
+	}
+	s.activePeersMtx.Unlock()
 
+	// If this is a newly initiated peerManager, we will create a
+	// peerHandler.
+	if !ok {
```

Comment:
just curious: how expensive is the old way w/ defer?

Reply:
not sure, could be the same I suppose. Actually, I'm curious if the lock is acquired inside the `go s.peerHandler()` below, could we potentially end up in a deadlock with the old way?